include/vkgcDefs.h

```diff
@@ -328,6 +328,8 @@ struct PipelineOptions {
                                    ///  then linking them, when possible.  When not possible this option is ignored.
   bool disableImageResourceCheck;  ///< If set, the pipeline shader will not contain code to check and fix invalid image
                                    ///  descriptors.
+  bool enableScratchBoundsCheck;   ///< If set, accessing a stack object out of bounds will be skipped for stores and
```

Comment:
Need to update the version history for this.
Is the intention that this be set default true in XGL to match current behaviour?
Should you also remove the targetFeatures += ",+enable-scratch-bounds-checks" code with this patch as well so that duplicate bounds checks are not generated?

Reply:
I tried to find something about out-of-bounds accesses in the SPIR-V and Vulkan specs but to no avail. The robustness extensions are only referring to buffer and image accesses but not about accessing a local (stack) array.
So I think an out-of-bounds OpLoad on a local variable is undefined behavior and we do not need to guard against it.

As the bounds checks incur a performance overhead, I would disable them by default and only enable them if needed.

Removing `+enable-scratch-bounds-checks` is included in this patch :)

Reply:
Thanks, I agree on no default enable.
Comment about updating version history in the header still stands.